pkg/odo/cli/cli.go

```diff
@@ -53,58 +64,17 @@ Additional help topics:{{range .Commands}}{{if .IsHelpCommand}}
 
 Use "{{.CommandPath}} [command] --help" for more information about a command.{{end}}
 `
+)
 
-// rootCmd represents the base command when called without any subcommands
-var rootCmd = &cobra.Command{
-	Use:   rootCommandName,
-	Short: "Odo (Openshift Do)",
-	Long: `Odo (OpenShift Do) is a CLI tool for running OpenShift applications in a fast and automated matter. Odo reduces the complexity of deployment by adding iterative development without the worry of deploying your source code.
-
-Find more information at https://github.com/redhat-developer/odo`,
-	Example: `  # Creating and deploying a Node.js project
-  git clone https://github.com/openshift/nodejs-ex && cd nodejs-ex
-  odo create nodejs
-  odo push
-
-  # Accessing your Node.js component
-  odo url create`,
-	PersistentPreRun: func(cmd *cobra.Command, args []string) {
-	},
-}
-
-// RootCmd exposes the root command to main package to allow inspection by completion code
-func RootCmd() *cobra.Command {
-	return rootCmd
-}
-
-// Execute adds all child commands to the root command and sets flags appropriately.
-// This is called by main.main(). It only needs to happen once to the rootCmd.
-func Execute() {
-
-	// checking the value of updatenotification in config
-	// before proceeding with fetching the latest version
-	cfg, err := config.New()
-	if err != nil {
-		util.CheckError(err, "")
-	}
-	if cfg.GetUpdateNotification() == true {
-		updateInfo := make(chan string)
-		go version.GetLatestReleaseInfo(updateInfo)
-
-		util.CheckError(rootCmd.Execute(), "")
-		select {
-		case message := <-updateInfo:
-			fmt.Println(message)
-		default:
-			glog.V(4).Info("Could not get the latest release information in time. Never mind, exiting gracefully :)")
-		}
-	} else {
-		util.CheckError(rootCmd.Execute(), "")
+// NewCmdOdo creates a new root command for odo
+func NewCmdOdo(name, fullName string) *cobra.Command {
```

Comment:
what is the difference between `name` and `fullName`? In all places where this function is called both parameters are the same

Reply:
@kadel For the `odo` command both are the same but for service create, `name` is `create` while full name is `odo service create` (i.e. the concatenation of all the parent command names). This is used in the usage/example strings.